lib/kamal/configuration/ssh.rb

```diff
@@ -35,7 +36,10 @@ def keys
   end
 
   def key_data
-    ssh_config["key_data"]
+    key_data = ssh_config["key_data"]
+    return unless key_data
+
+    key_data.map { |k| k.include?("-----BEGIN") ? k : secrets[k] }
```

Comment:
```suggestion
    key_data.map { |k| resolve_key_data(k) }
```

then put in private
```ruby
def resolve_key_data(key_or_name)
      # Try to parse as an SSH key first, if fail, we go old way.
      begin
        OpenSSL::PKey::RSA.new(key_or_name)
       key_or_name
      rescue OpenSSL::PKey::RSAError, TypeError
        # If parsing fails, treat as secret name
        secrets[key_or_name]
      end
 end
 ```
 
 PS: openssl is already available by sshkit